### PR TITLE
Strengthened Test Coverage for Organization Service

### LINKED ISSUE #4130: Improve Test Coverage for `organization.service` Package
## Description
The current test coverage for the `org.openelisglobal.organization.service` package is sitting at an overall instruction coverage of 23% (792 of 1,041 missed instructions) and a branch coverage of 23% (63 of 82 missed branches). 
Several service implementation classes currently lack adequate test suites, presenting a high risk for regressions.

### Coverage Breakdown by Class
As seen in the coverage report, the following service classes require immediate attention:

1. `OrganizationImportServiceImpl.java`: 0% Instruction / 0% Branch Coverage (114/115 lines missed, 8 methods missed)
2. `OrganizationExportServiceImpl.java`: 2% Instruction / 0% Branch Coverage (29/30 lines missed, 3 methods missed)
3. `OrganizationServiceImpl.java`: 65% Instruction / 73% Branch Coverage (21/70 lines missed, 10 methods missed)
4. `OrganizationTypeServiceImpl.java`: 76% Instruction Coverage (1/6 lines missed)
5. `OrganizationContactServiceImpl.java`: 100% Instruction Coverage (Fully covered)

#### Target Coverage Goals:

Bring overall package instruction and branch coverage above 70%.
Achieve comprehensive test coverage for critical components, specifically targeting OrganizationImportServiceImpl and OrganizationExportServiceImpl.

#### Tasks:
- [ ] Write unit tests for OrganizationImportServiceImpl to cover parsing, validation, and database import workflows.
- [ ] Expand and complete unit tests for OrganizationExportServiceImpl (covering active/inactive exports, endpoint mappings, and parent organization trees).
- [ ] Fill remaining test gaps in OrganizationServiceImpl and OrganizationTypeServiceImpl for edge cases and unexercised methods.

#### Definition of Done:
All new and existing unit/integration tests pass successfully in CI.
Coverage reports confirm the removal of zero-coverage indicators on import/export service implementations.

**Source of data:** Jacoco TeReport
Image

### PR DESCRIPTION
**Pull request by Ariho-Seth**
# Pull Requests Requirements

- [x] The PR title includes a brief description of the work done, including the
      Issue number if applicable.
- [ ] The PR includes a video showing the changes for the work done.
- [x] The PR title follows conventional commit label standards.
- [x] The changes confirm to the OpenElis Global x3
      [Styleguide and Design](https://uwdigi.atlassian.net/wiki/spaces/OG/pages/621346838/OpenELIS+Global+Style+Guide)
      documentation.
- [x] The changes include tests or are validated by existing tests.
- [x] I have read and agree to the Contributing
      [Guidelines](https://openelis-global.org/community/get-involved/) of this
      project.

## Summary

This PR includes improves the test coverage of the core organization service classes to enhance functionality and ensure stability within the organization module.

#### Changes Made

1. `OrganizationServiceImpl.java`: Updated business logic and service methods related to organization management, queries, and hierarchy handling.
2. `OrganizationTypeServiceImpl.java`: Refined methods and data handling for organization types.
3. `OrganizationExportServiceImpl.java`: Added unit tests 

## Related Issue

[Improve Test Coverage for the organization.service Package](https://github.com/DIGI-UW/OpenELIS-Global-2/issues/4130)

Closes #4130

### CODE REVIEW

**Review by pmanko — CHANGES_REQUESTED** [review]
Reviewed commit: `78639f1fe963881a2f5ce8b490bc0b35aa67602d`

Must fix before approval: this PR still does not test organization import, although #4130 requires it. Add those tests, or update the PR and issue scope so they do not claim that work. Also make the parent export test use stable FHIR IDs and verify that the bundle is built. The two smaller Copilot test-cleanup notes can be handled in the same update.

Verification: targeted organization import/export tests and the authoritative E2E checkpoint must pass.

File: src/test/java/org/openelisglobal/organisation/OrganizationExportServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+        when(fhirContext.newJsonParser()).thenReturn(fhirParser);
+        when(fhirParser.setPrettyPrint(true)).thenReturn(fhirParser);
+        when(fhirParser.encodeResourceToString(any())).thenReturn("{ \"resourceType\": \"Bundle\" }");
+
+        // When
+        String jsonResult = organizationExportService.exportFhirOrganizationsFromOrganizations(true);
+
+        // Then
+        assertNotNull(jsonResult);
```
Comment (Agaba-derrick) [inline src/test/java/org/openelisglobal/organisation/OrganizationExportServiceTest.java:65]: ideally i would opt for stronger assertion instead of using not null
i would assert specific scenerios  on the test

**Review by Agaba-derrick — CHANGES_REQUESTED** [review]

File: src/test/java/org/openelisglobal/organisation/OrganizationExportServiceTest.java (hunk 1 above)
Comment (Ariho-Seth) [reply src/test/java/org/openelisglobal/organisation/OrganizationExportServiceTest.java:65]: Sure thing, I've added stronger assertions

File: src/test/java/org/openelisglobal/organisation/OrganizationExportServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    private FhirContext fhirContext;
+    @Mock
+    private IParser fhirParser;
+    @InjectMocks
+    private OrganizationExportServiceImpl organizationExportService;
+
+    @Test
+    public void testExportFhirOrganizationsFromOrganizations_WhenActiveTrue() throws FhirTransformationException {
+        // Given
+        Organization organizationMock = new Organization();
```
[the remark this answers is not in the document]
Comment (Ariho-Seth) [reply src/test/java/org/openelisglobal/organisation/OrganizationExportServiceTest.java:44]: I've added unit tests for OrganizationImportServiceImpl

File: src/test/java/org/openelisglobal/organisation/OrganizationTypeServiceTest.java [hunk 3]
```diff
@@ -53,4 +53,12 @@ public void getOrganizationTypeByName() {
         assertEquals("Healthcare", organizationType2.getName());
         assertEquals("Healthcare Organization", organizationType2.getDescription());
     }
+
+    @Test
+    public void testGetOrganizationIdsForType_ShouldReturnTypesMatchingThePassedID() {
+        List<String> returnedTypes = organizationTypeService.getOrganizationIdsForType("1");
+        assertNotNull(returnedTypes);
```
Comment (Agaba-derrick) [inline src/test/java/org/openelisglobal/organisation/OrganizationTypeServiceTest.java:60]: assertNotNull(result) confirms something was returned, but says nothing about whether it's the right thing
can we get rid of it ans use an aggressive assertion

**Review by Agaba-derrick — CHANGES_REQUESTED** [review]
